### PR TITLE
Update Next.js example to monorepo style with shared code via linked pnpm workspace packages

### PR DESCRIPTION
**Pull request by gregmagolan**

### CODE REVIEW

File: next.js/packages/one/package.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "@nextjs-example/one",
+  "private": true,
+  "main": "src/main.js",
+  "dependencies": {
+    "@types/is-odd": "3.0.1",
```
Comment (alexeagle) [inline next.js/packages/one/package.json:6]: nit: we should set a good example by not overconstraining. You probably don't need any constraint on the version of this package, e.g. `*`

File: next.js/packages/one/node_modules/is-odd [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../node_modules/.pnpm/is-odd@3.0.1/node_modules/is-odd
```
Comment (alexeagle) [inline next.js/packages/one/node_modules/is-odd:1]: oops, node_modules needs gitignore?

File: next.js/packages/one/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+npm_link_all_packages(name = "node_modules")
+
+ts_project(
+    name = "one_ts",
+    srcs = ["src/main.ts"],
+    allow_js = True,
+    # runtime direct dependencies of the linked npm package
+    data = [
+        "//packages/one:node_modules/is-odd",
```
Comment (alexeagle) [inline next.js/packages/one/BUILD.bazel:13]: nit: why not just ":node_modules/is-odd"

fully qualifying seems like it makes the example look more complex

File: next.js/apps/alpha/package.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "private": true,
+  "dependencies": {
+    "@types/is-even": "1.0.0",
+    "is-even": "1.0.0"
```
Comment (alexeagle) [inline next.js/apps/alpha/package.json:5]: nit: overconstrained

**Review by alexeagle — APPROVED** [review]

File: next.js/packages/one/node_modules/is-odd (hunk 2 above)
Comment (gregmagolan) [reply next.js/packages/one/node_modules/is-odd:1]: 🤦

File: next.js/packages/one/BUILD.bazel (hunk 3 above)
Comment (gregmagolan) [reply next.js/packages/one/BUILD.bazel]: 🦕

File: next.js/packages/one/package.json (hunk 1 above)
Comment (gregmagolan) [reply next.js/packages/one/package.json:6]: 👍 I'll set it to `3.x.x` as good practice to constrain the major at least

File: next.js/apps/alpha/package.json (hunk 4 above)
Comment (gregmagolan) [reply next.js/apps/alpha/package.json:5]: 👍

**Pull request merged**